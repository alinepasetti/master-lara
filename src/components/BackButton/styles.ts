import styled, { css } from 'styled-components';
import { BiArrowBack } from 'react-icons/bi';

export const Back = styled(BiArrowBack)`${({ theme }) => css`
  width: 40px;
  height: 40px;
  color: ${theme.colors.mediumGray};
  background-color: ${theme.colors.white};
  font-size: 0.8rem;
  border-radius: 50%;
  padding: 0.7rem;
  position: absolute;
  left: 1.6rem;
  top: 1.6rem;
`}`;
