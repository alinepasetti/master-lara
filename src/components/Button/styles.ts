import styled, { css } from 'styled-components';

export const Button = styled.a` ${({ theme }) => css`
  width: 90%;
  display: block;
  padding: 1.6rem;
  font-size: 2.4rem;
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
  border: none;
  margin: 0 auto 2.4rem;
  text-align: center;
  text-decoration: none;
`}`;
