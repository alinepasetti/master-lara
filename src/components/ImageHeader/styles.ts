import styled from 'styled-components';

export const Image = styled.img`
  width: 100%;
  height: 40vh;
  object-fit: cover;
  object-position: center;
`;
