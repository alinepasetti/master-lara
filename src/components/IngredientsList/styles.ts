import styled, { css } from 'styled-components';
import { AiFillCloseCircle } from 'react-icons/ai';
import { AiFillCheckCircle } from 'react-icons/ai';

type HasIngredient = {
  hasIngredient: boolean;
};

const uncheckIconUrl = (theme) =>
  `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24' stroke-width='1.5' stroke='%23${theme.colors.secondaryColor.substring(
    1,
  )}' class='w-6 h-6'%3E%3Cpath stroke-linecap='round' stroke-linejoin='round' d='M9.75 9.75l4.5 4.5m0-4.5l-4.5 4.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z' /%3E%3C/svg%3E")`;

const checkIconUrl = (theme) =>
  `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24' stroke-width='1.5' stroke='%23${theme.colors.green.substring(
    1,
  )}' class='w-6 h-6'%3E%3Cpath stroke-linecap='round' stroke-linejoin='round' d='M9 12.75L11.25 15 15 9.75M21 12a9 9 0 11-18 0 9 9 0 0118 0z' /%3E%3C/svg%3E%0A")`;

export const Container = styled.ul`
  padding: 2.4rem;
  padding-top: 0;
  margin: 0;

  &> a {
    margin-top: 1.6rem;
  }
`;

export const ListItem = styled.li<HasIngredient>`${({
  theme,
  hasIngredient,
}) => css`
  list-style: none;

  &::before {
    display: inline-block;
    content: ${hasIngredient ? checkIconUrl(theme) : uncheckIconUrl(theme)};
    width: 1.8rem;
    padding-right: 1rem;
}
`}`;

export const CheckIcon = styled(AiFillCheckCircle)`${({ theme }) => css`
  color: ${theme.colors.green};
  margin-right: 2.4rem;
  height: 2.4rem;
  width: 2.4rem;
`}`;

export const UncheckIcon = styled(AiFillCloseCircle)`${({ theme }) => css`
  color: ${theme.colors.secondaryColor};
  margin-right: 2.4rem;
  height: 2.4rem;
  width: 2.4rem;
`}`;
