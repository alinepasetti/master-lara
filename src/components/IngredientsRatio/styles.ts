import styled, { css } from 'styled-components';

export const Container = styled.p`${({ theme }) => css`
  font-size: 1.6rem;
  margin: 0 0;
  color: ${theme.colors.white};
  background-color: ${theme.colors.primaryColor};
  font-size: 1.2rem;
  font-weight: 400;
  position: absolute;
  right: 1.6rem;
  top: 1.6rem;
  border-radius: 8px;
  box-shadow: 0 0 5px ${theme.colors.mediumGray};
  padding: 1rem;
`}`;
