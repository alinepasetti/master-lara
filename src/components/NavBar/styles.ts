import styled from 'styled-components';

export const NavBar = styled.nav`
  width: 100%;
  padding: 1rem 1.6rem;
  background-color: black;
`;

export const Image = styled.img`
  width: 20%;
  display: block;
  margin: 0 auto;
`;
