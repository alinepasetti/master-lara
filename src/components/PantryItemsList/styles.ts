import styled, { css } from 'styled-components';

type HasIngredient = {
  hasIngredient?: boolean;
};

export const Container = styled.ul`
  height: 100%;
  width: 100%;
  min-height: 4.0rem;
  padding: 0 0.8rem;
  margin: 1.6rem 0;
  display: flex;
  justify-content: space-around;
  flex: 1;
  flex-wrap: wrap;
  list-style: none;
`;

export const PantryItem = styled.li<HasIngredient>`${({ theme }) => css`
  color: ${theme.colors.primaryColor};
  background-color: ${theme.colors.white};
  padding: 1.25rem;
  width: 47%;
  border-radius: 8px;
  margin: 0.8rem 0;
  & > span {
    display: flex;
    justify-content: space-between;
  }

  & > span > svg {
    width: 2.5rem;
  }

  @media screen and (min-width: 768px) {
    width: 31.5%;
  }

  @media screen and (min-width: 1164px) {
    width: 23%;
  }
`}`;

export const Message = styled.p`${({ theme }) => css`
  font-size: 1.6rem;
  width: 100%;
  text-align: center;
  margin: 0.8rem 0;
  color: ${theme.colors.primaryColor};
`}`;
