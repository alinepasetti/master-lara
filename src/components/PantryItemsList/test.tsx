import { screen, within } from '@testing-library/react';
import { PantryItemsList } from '.';
import { renderTheme } from '../../styles/render-theme';
import { PantryContext } from '../../contexts/PantryProvider/context';
import { buildPantryProviderValueMock } from '../../mocks/pantry';
import userEvent from '@testing-library/user-event';
import { vi } from 'vitest';

const pantryItemsListElement = (
  pantryItems?: string[],
  searchValue?: string,
  removePantryItemMock?,
) => (
  <PantryContext.Provider
    value={buildPantryProviderValueMock(
      pantryItems,
      searchValue,
      removePantryItemMock,
    )}
  >
    <PantryItemsList />
  </PantryContext.Provider>
);

describe('<PantryItemsList />', () => {
  it('should render 1 list item', () => {
    renderTheme(pantryItemsListElement(['banana']));
    const list = screen.getByRole('list', {
      name: /ingredients/i,
    });

    const { getAllByRole } = within(list);
    const items = getAllByRole('listitem');
    expect(items.length).toBe(1);
  });

  it('should render list of ingredients added by the user', () => {
    renderTheme(pantryItemsListElement(['sugar', 'cane']));
    const list = screen.getByRole('list', {
      name: /ingredients/i,
    });
    const { getAllByRole } = within(list);
    const items = getAllByRole('listitem');
    const ingredients = items.map((item) => item.textContent);

    expect(ingredients).toEqual(['sugar', 'cane']);
  });

  it('should call function to remove an element when clicked', () => {
    const removePantryItemMock = vi.fn();

    renderTheme(
      pantryItemsListElement(
        ['sugar', 'cane'],
        undefined,
        removePantryItemMock,
      ),
    );
    const closeIcon = screen.getByText('sugar').lastElementChild;

    userEvent.click(closeIcon);

    expect(removePantryItemMock).toBeCalledTimes(1);
  });

  it('should match snapshot', () => {
    renderTheme(pantryItemsListElement());
    expect(screen.getByText('No items on the pantry yet.')).toBeInTheDocument();
  });

  it('should match snapshot with no items', () => {
    renderTheme(pantryItemsListElement());
    expect(
      screen.getByRole('list', { name: /ingredients/i }),
    ).toMatchSnapshot();
  });

  it('should match snapshot with items', () => {
    renderTheme(pantryItemsListElement(['sugar', 'cane']));
    expect(
      screen.getByRole('list', { name: /ingredients/i }),
    ).toMatchSnapshot();
  });
});
