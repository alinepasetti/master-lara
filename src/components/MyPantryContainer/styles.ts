import styled from 'styled-components';
import { Button as SearchButton } from '../Button/styles';

export const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;

  ${SearchButton} {
    border-radius: 8px;
    width: 10.8rem;
  }
`;
