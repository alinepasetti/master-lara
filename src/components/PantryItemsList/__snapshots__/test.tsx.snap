// Vitest Snapshot v1

exports[`<PantryItemsList /> > should match snapshot with items 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  min-height: 4.0rem;
  padding: 0 0.8rem;
  margin: 1.6rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
}

.c1 {
  color: #000000;
  background-color: #FFFFFF;
  padding: 1.25rem;
  width: 47%;
  border-radius: 8px;
  margin: 0.8rem 0;
}

.c1 > span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 > span > svg {
  width: 2.5rem;
}

.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

<ul
  aria-label="ingredients-list"
  class="c0"
>
  <li
    class="c1"
  >
    <span>
      sugar
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
        />
      </svg>
    </span>
  </li>
  <li
    class="c1"
  >
    <span>
      cane
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
        />
      </svg>
    </span>
  </li>
</ul>
`;

exports[`<PantryItemsList /> > should match snapshot with no items 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  min-height: 4.0rem;
  padding: 0 0.8rem;
  margin: 1.6rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
}

.c1 {
  font-size: 1.6rem;
  width: 100%;
  text-align: center;
  margin: 0.8rem 0;
  color: #000000;
}

<ul
  aria-label="ingredients-list"
  class="c0"
>
  <p
    class="c1"
  >
    No items on the pantry yet.
  </p>
</ul>
`;
